src/main/scala/is/hail/utils/SummaryCombiner.scala

```diff
@@ -0,0 +1,70 @@
+package is.hail.utils
+
+import is.hail.annotations.Annotation
+import is.hail.variant.Variant
+
+import scala.collection.mutable
+
+case class SummaryResult(samples: Int, variants: Long, callRate: Option[Double],
+  contigs: Set[String], multiallelics: Long, snps: Long, indels: Long,
+  complex: Long, mostAlleles: Int)
+
+class SummaryCombiner[T](f: Iterable[T] => Int) extends Serializable {
+  private val contigs = mutable.Set.empty[String]
+  private var nCalled = 0L
+  private var nVariants = 0L
+  private var multiallelics = 0L
+  private var mostAlleles = 2
+  private var snps = 0L
+  private var indels = 0L
+  private var complex = 0L
+
+  def merge(other: SummaryCombiner[T]): SummaryCombiner[T] = {
+
+    contigs ++= other.contigs
+    nCalled += other.nCalled
+    nVariants += other.nVariants
+    multiallelics += other.multiallelics
+    mostAlleles = math.max(mostAlleles, other.mostAlleles)
+    snps += other.snps
+    indels += other.indels
```

Comment:
Split out in, del and mnp.  These fields should conceptually be "count" on the AltAlellele type.